src/rgw/driver/rados/rgw_notify.cc

```diff
@@ -161,17 +178,41 @@ class Manager : public DoutPrefixProvider {
     }   
   };
 
+  enum class EntryProcessingResult {
+    Failure, Successful, Sleeping, Expired
+  };
+
   // processing of a specific entry
   // return whether processing was successfull (true) or not (false)
-  bool process_entry(const cls_queue_entry& entry, yield_context yield) {
+  EntryProcessingResult process_entry(const Manager& manager, persistency_tracker& entry_persistency_tracker,
```

Comment:
we dont really need to pass the manager here, we just need `cct`, or evewn better, the 2 configured values